pages/items/[id]/index.js

```diff
@@ -0,0 +1,40 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+
+import ItemDetails from "@/components/ItemDetails";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const { isReady } = router;
+  const { id } = router.query;
+
+  const { data: item, mutate, isLoading, error } = useSWR(`/api/items/${id}`);
+
+  async function handleStatus() {
+    const newItem = { ...item, inDiscuss: !item.inDiscuss };
+    await fetch(`/api/items/${id}`, {
+      method: "PUT",
+      body: JSON.stringify(newItem),
+    });
+    mutate(newItem);
+  }
+
+  if (!isReady || isLoading) {
```

Comment:
Very nice check!

Reply:
Nice solution!

Reply:
Sorry about my first comment. Using isLoading like this (for conditional rendering) is perfectly fine! But router.isReady should actually only be used in a useEffect and not for conditional rendering, see the [Next.JS router docs](https://nextjs.org/docs/api-reference/next/router#router-object)